hypershift/hosted-wrapper.py

```diff
@@ -256,59 +268,65 @@ def get_metadata(kubeconfig, my_path, duration, cluster_name, uuid, operation):
+    while True:
         logging.debug(cmd)
         process = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True, env=myenv)
         stdout, stderr = process.communicate()
-        current_cluster_count = 0
-        state = {}
-        error = []
-        # Count the various states/status' and report it to logging
-        for line in stdout.splitlines():
-            if cluster_name_seed in line:
-                if len(line.split()) >= 3:
-                    current_cluster_count += 1
-                    state_key = line.split()[3] if 'Completed' in line else line.split()[2]
-                    state[state_key] = state.get(state_key, 0) + 1
-                    # Not sure about the error state key
-                    if state_key == "error":
-                        error.append(line.split()[0])
-
-        logging.info('Requested Clusters for test %s: %d' % (my_uuid, cluster_count))
-        if current_cluster_count != 0:
-            logging.debug(state.items())
-            state_output = "Current clusters state: " + str(current_cluster_count) + " clusters"
-            for i in state.items():
-                state_output += " (" + str(i[0]) + ": " + str(i[1]) + ")"
-            logging.info(state_output)
-            if error:
-                logging.warning('Clusters in error state: %s' % error)
-            clusters_resume['state'] = state
-            clusters_resume['clusters_created'] = current_cluster_count
-        time.sleep(delay)
+        installing_clusters = 0
+        installed_clusters = 0
+        other_clusters = 0
+        other_list = []
+        for cluster in json.loads(stdout)['items']:
+            if cluster_name_seed in cluster['metadata']['name']:
+                status = cluster['status']['version']['history'][0]
+                if status['state'] == 'Completed':
+                    installed_clusters += 1
+                elif status['state'] == 'Partial':
+                    installing_clusters += 1
+                else:
+                    other_clusters += 1
+                    other_list.append(cluster['metadata']['name'])
+        logging.info('Requested Clusters for test %s: %d of %d' % (my_uuid, installing_clusters + installed_clusters + other_clusters, cluster_count))
+        logging.info('   Installing: %d' % installing_clusters)
+        logging.info('   Ready: %d' % installed_clusters)
+        logging.info('   Other: %d (%s)' % (other_clusters, other_list))
+        if installed_clusters == cluster_count:
+            with all_clusters_installed:
+                logging.info('All requested clusters on ready status, notifying threads to start e2e-benchmarking processes')
+                all_clusters_installed.notify_all()
+            break
+        elif installed_clusters + other_clusters == cluster_count and installing_clusters == 0:
+            logging.error('No cluster pending to install and not all clusters are ready. Exiting...')
```

Comment:
Is there any useful info we could dump if we get here?

Reply:
List of clusters which are on "other" status is already printed on the logging, 5 lines over this

Reply:
ack. that works